vllm/v1/spec_decode/eagle.py

```diff
@@ -103,13 +103,6 @@ def __init__(
                 and not self.speculative_config.enforce_eager
             )
 
-        self.cudagraph_batch_sizes = (
-            (sorted(self.vllm_config.compilation_config.cudagraph_capture_sizes))
-            if self.use_cuda_graph
-            else []
-        )
-
-        self.use_cuda_graph = self.use_cuda_graph and bool(self.cudagraph_batch_sizes)
```

Comment:
What is the value of `self.compilation_config.max_cudagraph_capture_size` when there aren't any valid sizes? We had to add this check to specifically cover the case when `self.vllm_config.compilation_config.cudagraph_capture_sizes` is empty but `use_cuda_graph` is True. Will changing the heuristic to `max_cudagraph_capture_size` keep this fixed?